kolibri/plugins/user/assets/src/views/sign-in-page/facility-modal.vue

```diff
@@ -0,0 +1,85 @@
+<template>
+
+  <core-modal @cancel="emitClose" title="Select a facility">
+    <form @submit.prevent="submitAndClose">
+
+    </form>
+    Which facility do you want to log into?
+
+    <k-radio-button
+      v-for="facility in facilities"
+      v-model="selectedFacility"
+      :key="facility.id"
+      :label="facility.name"
+      :radiovalue="facility.id"
+    />
+
+    <k-button
+      class="core-modal-buttons"
+      @click="emitClose"
+      text="Cancel"
+    />
+    <k-button
+      class="core-modal-buttons"
+      text="Continue"
+      :primary="true"
+      type="submit"
+    />
+
+
+  </core-modal>
+
+</template>
+
+
+<script>
+
+  import coreModal from 'kolibri.coreVue.components.coreModal';
+  import kRadioButton from 'kolibri.coreVue.components.kRadioButton';
+  import kButton from 'kolibri.coreVue.components.kButton';
+  import { getFacilityConfig } from 'kolibri.coreVue.vuex.actions';
+
+  export default {
+    name: 'facilityModal',
+    components: {
+      coreModal,
+      kRadioButton,
+      kButton,
+    },
+    props: {
+      facilities: {
+        type: Array,
+        required: true,
+      },
```

Comment:
Since this component is already referencing vuex, might as well just reference `state.core.facilities` directly in this component. Passing it as a prop with type `Array` creates extra complexity with no benefit.

Reply:
I think I have a hard time striking a balance between making a component completely functional vs state driven

Reply:
compromise can result in inconsistency